scanners/nessus/nessus_none.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import time
+from dataclasses import dataclass
+from dataclasses import field
+from typing import List
+from typing import Optional
+
+import dacite
+import requests.exceptions
+from py_nessus_pro import PyNessusPro
+
+from configmodel import RapidastConfigModel
+from scanners import RapidastScanner
+from scanners import State
+
+
+@dataclass
+class NessusServerConfig:
+    url: str
+    username: str
+    password: str
+
+
+@dataclass
+class NessusScanConfig:
+    name: str
+    policy: str
+    targets: List[str]
+    folder: str = field(default="rapidast")
+
+    def targets_as_str(self) -> str:
+        return " ".join(self.targets)
+
+
+@dataclass
+class NessusConfig:
+    server: NessusServerConfig
+    scan: NessusScanConfig
+
+
+# XXX required by ./rapidast.py
+CLASSNAME = "Nessus"
+
+END_STATUSES = [
+    "completed",
+    "canceled",
+    "imported",
+    "aborted",
+]
+
+
+class Nessus(RapidastScanner):
+    def __init__(self, config: RapidastConfigModel, ident: str = "nessus"):
+        super().__init__(config, ident)
+        self._nessus_client: Optional[PyNessusPro] = None
+        self._scan_id: Optional[int] = None
+        nessus_config_section = config.subtree_to_dict(f"scanners.{ident}")
+        if nessus_config_section is None:
+            raise ValueError("'scanners.nessus' section not in config")
+        # self.auth_config = config.subtree_to_dict("general.authentication")  # creds for target hosts
+        # XXX self.config is already a dict with raw config values
+        self.cfg = dacite.from_dict(data_class=NessusConfig, data=nessus_config_section)
+        self.sleep_interval: int = 20
+        self._connect()
+
+    def _connect(self):
+        logging.debug(f"Connecting to nessus instance at {self.cfg.server.url}")
+        try:
+            self._nessus_client = PyNessusPro(
+                self.cfg.server.url,
+                self.cfg.server.username,
+                self.cfg.server.password,
+                log_level="debug",
+            )
+        except requests.exceptions.RequestException as e:
+            logging.error(f"Failed to connect to {self.cfg.server.url}: {e}")
+            raise
+
+    @property
+    def nessus_client(self) -> PyNessusPro:
+        if self._nessus_client is None:
+            raise RuntimeError(f"Nessus client not connected: {self.state}")
+        return self._nessus_client
+
+    @property
+    def scan_id(self) -> int:
+        if self._scan_id is None:
+            raise RuntimeError("scan_id is None")
+        return self._scan_id
+
+    def setup(self):
+        logging.debug(f"Creating new scan named {self.cfg.scan.folder}/{self.cfg.scan.name}")
+        self._scan_id = self.nessus_client.new_scan(
```

Comment:
any chance new_scan() could fail and return -1(https://github.com/Matbe34/py-nessus-pro/blob/f49a60643c7d553ae9ae8172f6454028fafc852e/py_nessus_pro/py_nessus_pro.py#L100)? then handling the exception would be necessary.

Reply:
Not 100% sure if I understood correctly, but I added an error check which will raise an exception if the scan_id is less than 0.